pkg/services/live/pushws/push.go

```diff
@@ -101,15 +102,10 @@ const (
 )
 
 func (s *Handler) ServeHTTP(rw http.ResponseWriter, r *http.Request) {
-	var streamID string
-
-	streamID = r.Header.Get("X-Grafana-Live-Stream")
-	if streamID == "" {
-		streamID = r.URL.Query().Get("gf_live_stream")
-	}
-	if streamID == "" {
+	streamID, ok := livecontext.GetContextStreamID(r.Context())
+	if !ok || streamID == "" {
 		logger.Warn("Push request without stream ID")
-		rw.WriteHeader(http.StatusBadRequest)
+		rw.WriteHeader(http.StatusInternalServerError)
```

Comment:
why not bad request?

Reply:
This can happen if we have not put streamId into context - so a developer mistake.